crates/svm-cli/src/app_tx.rs

```diff
@@ -0,0 +1,79 @@
+use std::error::Error;
+use std::fs::File;
+use std::io::prelude::{Read, Write};
+
+use crate::common;
+use svm_app::{
+    raw::{decode_exec_app, NibbleIter},
+    testing::AppTxBuilder,
+    types::{AppTransaction, WasmValue},
+};
+
+pub fn encode(
+    version: u32,
+    app_addr_hex: &str,
+    func_idx: u16,
+    func_buf_hex: Option<&str>,
+    func_args: Option<Vec<&str>>,
+    output_path: &str,
+) -> Result<usize, Box<dyn Error>> {
+    let app_addr = common::decode_addr(app_addr_hex)?;
+
+    let mut func_buf = None;
+    if let Some(func_buf_hex) = func_buf_hex {
+        func_buf = Some(common::decode_hex(func_buf_hex)?);
+    }
+
+    let mut func_args_vals: Option<Vec<WasmValue>> = None;
+    if let Some(func_args) = func_args {
+        func_args_vals = Some(common::decode_args(func_args)?);
+    }
+
+    let mut builder = AppTxBuilder::new()
+        .with_version(version)
+        .with_app(&app_addr.into())
+        .with_func_index(func_idx);
+    if let Some(func_buf) = func_buf {
+        builder = builder.with_func_buf(&func_buf)
+    }
+    if let Some(func_args_vals) = func_args_vals {
```

Comment:
`func_args` are optional:
https://github.com/spacemeshos/svm/blob/master/crates/svm-app/src/testing/app_tx_builder.rs#L99

Reply:
My code is already treating it as optional, no?

Reply:
please try:
```rust
builder.with_func_args(&func_args.unwrap_or(vec![]))
```